### DISCUSSION TITLE
classes/class_editorscenepostimport

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#166] [User-contributed notes]
# classes/class_editorscenepostimport

Inherits: RefCounted< Object Post-processes scenes after import. Description: Imported scenes can be automatically modified right after import by setting their Custom Script Import property to a to...

https://docs.godotengine.org/en/stable/classes/class_editorscenepostimport.html

**Comment by rozab** [comment]
If you are fiddling with the node hierarchy and your script fails with:
```
scene/main/node.cpp:2518 - Condition "!is_ancestor_of(p_node)" is true. Returning: false
```
it could be that some of your node owners are set to a node which is no longer in the scene, causing this error when Godot attempts to pack the scene.

This can be resolved by adding this code to your script:

```gdscript
@tool
extends EditorScenePostImport

func _post_import(scene):

    # hierarchy fiddling...

    set_children_owner(new_scene, new_scene)
    return new_scene

func set_children_owner(node, owner):
	for n in node.get_children():
		n.owner = owner
		set_children_owner(n, owner)
```

**Comment by BManx2000** [comment]
If your script modifies animations and "save to file" is enabled, it will not work because the animations are saved before _post_import runs. Workaround is to use an EditorPlugin as described here: https://github.com/godotengine/godot/issues/85738#issuecomment-2558166987
